genesyscloud/telephony_providers_edges_phone/resource_genesyscloud_telephony_providers_edges_phone_test.go

```diff
@@ -532,3 +530,26 @@ func deleteDidPool(config *platformclientv2.Configuration, id string) error {
 	}
 	return nil
 }
+func deleteDidPoolWithNumber(number string) {
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+		didPools, _, getErr := edgesAPI.GetTelephonyProvidersEdgesDidpools(pageSize, pageNum, "", nil)
+		if getErr != nil {
+			return
+		}
+
+		if didPools.Entities == nil || len(*didPools.Entities) == 0 {
+			break
+		}
+
+		for _, didPool := range *didPools.Entities {
+			if (didPool.StartPhoneNumber != nil && *didPool.StartPhoneNumber == number) ||
+				(didPool.EndPhoneNumber != nil && *didPool.EndPhoneNumber == number) {
+				edgesAPI.DeleteTelephonyProvidersEdgesDidpool(*didPool.Id)
+				time.Sleep(5 * time.Second)
+			}
+		}
+	}
```

Comment:
In another PR, a dev made a good point that its a bad idea to paginate and delete entities at the same time because the pages are going to change as you paginate through them. It's better to loop through the entities and add all to a list, and then after all have been collected that match the condition, loop through them and do the deletions

Reply:
+1

Reply:
Modified it to collect list of didpool first, then delete.